### PR TITLE
Implement JSON-RPC `AccountService` and reorganize TCK code to use service-oriented architecture

### LINKED ISSUE #671: TCK-CPP: Implement `Account` JSON-RPC Service
### Problem

At the moment endpoints for any kind of Hedera related transactions and queries are being held by the `sdkClient`. We should  introduce endpoint grouping. 

### Solution

We should create an `Account` JSON-RPC Service to hold `Account` specific endpoints.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by rwalworth**
This PR reorganizes and cleans up the TCK server code to provide a more modular and organized architecture.

- Implement `AccountService`, which houses all crypto-service based JSON-RPC methods.
- Reorganize `SdkClient` to house client-based JSON-RPC methods (`setup` and `reset`).
- Create `KeyService`, which houses the `generateKey` JSON-RPC method as well as helper functions to be used by the rest of the TCK server codebase.
- Create parameter structures for all JSON-RPC methods and automatic JSON parsing functionality for cleaner function declarations.
- Remove `JsonTypeMapper.h` as its primary functionality of parsing incoming JSON has been replaced by the parameter structs.

**Related issue(s)**:

Fixes #671 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/tck/CMakeLists.txt
Comment (gsstoykov) [inline src/tck/CMakeLists.txt:1]: What are your thoughts on having TCK build with a flag in the CMake config? Do we intend to have the TCK built every time we build the SDK itself?

**Review by gsstoykov — DISMISSED** [review]
LGTM. Just a few points for discussion.

**Review by jeromy-cannon — APPROVED** [review]

File: src/tck/CMakeLists.txt
Comment (rwalworth) [reply src/tck/CMakeLists.txt:1]: I've thought about this before with our unit/integration tests as well, I think it'd be smart to make it configurable. I'll make a separate issue for this.

Comment (rwalworth) [reply src/tck/CMakeLists.txt:1]: #816

**Review by gsstoykov — APPROVED** [review]

File: src/tck/CMakeLists.txt
Comment (gsstoykov) [reply src/tck/CMakeLists.txt:1]: maybe I can pick it up

**Pull request merged**